diff --git a/board/renesas/rzv2m-dev/rzv2m-dev.c b/board/renesas/rzv2m-dev/rzv2m-dev.c
index 38561025..1baf6eed 100644
--- a/board/renesas/rzv2m-dev/rzv2m-dev.c
+++ b/board/renesas/rzv2m-dev/rzv2m-dev.c
@@ -409,6 +409,33 @@ int board_early_init_f(void)
 	/* HDMI phy setting : SYS GEN_REG2 reg : offset 208h */
 	CMN_REG_Write32(0xA3F03208,0x60106010);
 
+	/*- Release USB HUB reset : PFC_P06_GPIO_D06 -*/
+	/* PORT06 Input/Output Enable Mask Register */
+	CMN_REG_Write32(0xB62501b4, 0x0FFF0000);
+	 
+	/* PORT06 Output control register */
+	CMN_REG_Write32(0xB6250184, 0x00400040);
+ 	
+	/* PORT06 Output value control 
+	 *  set RST to HIGH
+	 */
+	CMN_REG_Write32(0xB6250180, 0x00400040);
+	
+	/*- Enable SDIO Power select SDI0 PWR to 3V3
+	 * PFC_P09_GPIO_D4 - SD0 PWR Select
+	 * SD0 PWR Enable is given bt the power sequencer MCU
+	 */
+	/* PORT09 Input/Output Enable Mask Register */
+	CMN_REG_Write32(0xB6250274, 0x00FF0000);
+	 
+	/* PORT09 Output control register */
+	CMN_REG_Write32(0xB6250244, 0x00100010);
+	
+	/* PORT09 Output value control
+	 * set LOW to P0604 for 3V3
+	 */
+	CMN_REG_Write32(0xB6250240, 0x00100000);
+
 	return 0;
 }
 
