--- ./lib/include/openamp/virtqueue.h.org	2021-06-08 19:37:24.803672663 +0900
+++ ./lib/include/openamp/virtqueue.h	2021-06-08 19:47:50.356084152 +0900
@@ -57,6 +57,11 @@
 	int len;
 };
 
+struct vq_desc_extra {
+	void *cookie;
+	uint16_t ndescs;
+};
+
 struct virtqueue {
 	struct virtio_device *vq_dev;
 	const char *vq_name;
@@ -99,10 +104,7 @@
 	 * Other fields in this structure are not used currently.
 	 */
 
-	struct vq_desc_extra {
-		void *cookie;
-		uint16_t ndescs;
-	} vq_descx[0];
+	struct vq_desc_extra vq_descx[0];
 };
 
 /* struct to hold vring specific information */
