diff --git a/arch/arm64/boot/dts/renesas/r9a09g011gbg.dtsi b/arch/arm64/boot/dts/renesas/r9a09g011gbg.dtsi
index b961e0d63..907eaaa0e 100644
--- a/arch/arm64/boot/dts/renesas/r9a09g011gbg.dtsi
+++ b/arch/arm64/boot/dts/renesas/r9a09g011gbg.dtsi
@@ -564,6 +564,7 @@ GIC_SPI 278 IRQ_TYPE_LEVEL_HIGH>,
             phy0: ethernet-phy@0 {
                 rxc-skew-ps = <1500>;
                 reg = <0>;
+                reset-gpios = <&gpio6 11 GPIO_ACTIVE_LOW>;
             };
         };
 
